ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Exposes stats related to a {@link BlobStore} that is useful to different components.
+ *
+ * Note: This is the v0 implementation of BlobStoreStats. The v0 implementation walks through the entire index
+ * and collect data needed to serve stats related requests.
+ */
+public class BlobStoreStats implements StoreStats {
+  static final String IO_SCHEDULER_JOB_TYPE = "BlobStoreStats";
+  static final String IO_SCHEDULER_JOB_ID = "indexSegment_read";
+
+  private final PersistentIndex index;
+  private final Time time;
+  private final DiskIOScheduler diskIOScheduler;
+
+  BlobStoreStats(PersistentIndex index, Time time, DiskIOScheduler diskIOScheduler) {
+    this.index = index;
+    this.time = time;
+    this.diskIOScheduler = diskIOScheduler;
+  }
+
+  // TODO remove this once #541 is merged
+  public static String getServiceId(StoreKey key) {
+    return key.getID().substring(0, 1);
+  }
+
+  // TODO remove this once #541 is merged
+  public static String getContainerId(StoreKey key) {
+    return key.getID().substring(key.getID().length() - 1);
+  }
+
+  @Override
```

Comment:
Correct me if I'm wrong but I don't see the point of `{@inheritDoc}` if there are no additional comments that needs to be added to the interface's Javadoc. The `ctrl + J` command to view the Javadoc works fine even without `{@inheritDoc}`.